
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="</br></br>Select a tutorial and learn how to access data from the ICDC and perform a bioinformatic analysis">
      
      
      
        <link rel="canonical" href="https://cbiit.github.io/icdc-bioinformatics/Tutorials/RNA-Seq-on-CGC/rna_seq_04/">
      
      <link rel="icon" href="../../../images/canine_bubble.png">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.2.11">
    
    
      
        <title>Analysis within the Cancer Genomics Cloud (CGC) - The Integrated Canine Data Commons Bioinformatics Training Center</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.8c5ef100.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.9647289d.min.css">
        
          
          
          <meta name="theme-color" content="">
        
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../stylesheets/extra.css">
    
      <link rel="stylesheet" href="../../../stylesheets/style.css">
    
    <script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="custom" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#analysis-within-the-cancer-genomics-cloud-cgc" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="The Integrated Canine Data Commons Bioinformatics Training Center" class="md-header__button md-logo" aria-label="The Integrated Canine Data Commons Bioinformatics Training Center" data-md-component="logo">
      
  <img src="../../../assets/canine_bubble.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            The Integrated Canine Data Commons Bioinformatics Training Center
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Analysis within the Cancer Genomics Cloud (CGC)
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/CBIIT/icdc-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../../.." class="md-tabs__link">
      Home
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../../about/" class="md-tabs__link">
      About
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../rna_seq_01/" class="md-tabs__link md-tabs__link--active">
        Tutorials
      </a>
    </li>
  

  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="The Integrated Canine Data Commons Bioinformatics Training Center" class="md-nav__button md-logo" aria-label="The Integrated Canine Data Commons Bioinformatics Training Center" data-md-component="logo">
      
  <img src="../../../assets/canine_bubble.png" alt="logo">

    </a>
    The Integrated Canine Data Commons Bioinformatics Training Center
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/CBIIT/icdc-bioinformatics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../about/" class="md-nav__link">
        About
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Tutorials
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Tutorials" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Tutorials
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" type="checkbox" id="__nav_3_1" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3_1">
          RNA-Seq on the CGC
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="RNA-Seq on the CGC" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_1">
          <span class="md-nav__icon md-icon"></span>
          RNA-Seq on the CGC
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_01/" class="md-nav__link">
        RNA-Seq Tutorial Overview
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_02/" class="md-nav__link">
        Introduction to RNA-Seq
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_03/" class="md-nav__link">
        Building a Cohort within the ICDC
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Analysis within the Cancer Genomics Cloud (CGC)
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Analysis within the Cancer Genomics Cloud (CGC)
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-1-copy-tools-into-project" class="md-nav__link">
    Step 1: Copy tools into project
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-inspect-summary-stats-for-each-file-with-sambamba-flagstat" class="md-nav__link">
    Step 2: Inspect summary stats for each file with Sambamba Flagstat
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-inspect-bam-file-headers-with-samtools-view" class="md-nav__link">
    Step 3: Inspect bam file headers with Samtools View
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-load-the-appropriate-reference-annotation-file" class="md-nav__link">
    Step 5: Load the appropriate reference annotation file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-count-sequencing-reads-with-htseq-count" class="md-nav__link">
    Step 4: Count sequencing reads with htseq-count
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-create-a-csv-file-with-phenotype-data-for-all-samples-for-deseq2" class="md-nav__link">
    Step 5: Create a csv file with phenotype data for all samples for DESeq2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-conduct-differential-expression-with-deseq2" class="md-nav__link">
    Step 6: Conduct differential expression with DESeq2
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-1-copy-tools-into-project" class="md-nav__link">
    Step 1: Copy tools into project
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-inspect-summary-stats-for-each-file-with-sambamba-flagstat" class="md-nav__link">
    Step 2: Inspect summary stats for each file with Sambamba Flagstat
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-inspect-bam-file-headers-with-samtools-view" class="md-nav__link">
    Step 3: Inspect bam file headers with Samtools View
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-load-the-appropriate-reference-annotation-file" class="md-nav__link">
    Step 5: Load the appropriate reference annotation file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-count-sequencing-reads-with-htseq-count" class="md-nav__link">
    Step 4: Count sequencing reads with htseq-count
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-create-a-csv-file-with-phenotype-data-for-all-samples-for-deseq2" class="md-nav__link">
    Step 5: Create a csv file with phenotype data for all samples for DESeq2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-conduct-differential-expression-with-deseq2" class="md-nav__link">
    Step 6: Conduct differential expression with DESeq2
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/CBIIT/icdc-bioinformatics/edit/master/docs/Tutorials/RNA-Seq-on-CGC/rna_seq_04.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="analysis-within-the-cancer-genomics-cloud-cgc">Analysis within the Cancer Genomics Cloud (CGC)</h1>
<h2 id="step-1-copy-tools-into-project">Step 1: Copy tools into project</h2>
<ul>
<li>Click on Public Apps from the top navigation bar to expand the dropdown menu</li>
<li>Select Workflows and Tools</li>
<li>Click the Browse buttons app</li>
<li>
<p>Use the search bar to find each tool listed below</p>
<ul>
<li>Sambamba Flagstat</li>
<li>Samtools View</li>
<li>Samtools Sort</li>
<li>HTSeq-count</li>
<li>DESeq2</li>
</ul>
</li>
<li>
<p>For each tool, click the Copy button and then Select the appropriate project from the dropdown menu and then click the Copy button once more</p>
</li>
<li>Click on Projects from the main navigation bar to expand the dropdown menu and select the appropriate project name to return to the project</li>
<li>Click on Apps from the secondary navigation bar to ensure all tools were effectively copied into the project
<img alt="Tools loaded" src="../rna-seq-images/cgc-apps-loaded.png" title="Tools loaded" /></li>
</ul>
<h2 id="step-2-inspect-summary-stats-for-each-file-with-sambamba-flagstat">Step 2: Inspect summary stats for each file with Sambamba Flagstat</h2>
<ul>
<li>Click on Sambamba Flagstat</li>
<li>Click on the Run button from the top right-hand side</li>
<li>Click on the toggle to turn Batching On</li>
<li>Select File from the Batch by dropdown menu</li>
<li>Click on Input alignments</li>
<li>Select all bam files by clicking on the respective checkboxes</li>
<li>Click on the Save selection button on the top right-hand side</li>
<li>Click on the Run button on the top right-hand side 
<img alt="Samtools Flagstat" src="../rna-seq-images/cgc-flagstat-results.png" title="Samtools Flagstat" /></li>
</ul>
<h2 id="step-3-inspect-bam-file-headers-with-samtools-view">Step 3: Inspect bam file headers with Samtools View</h2>
<ul>
<li>Click on Apps from the secondary navigation menu </li>
<li>Click on Samtools View</li>
<li>Click on the Run button from the top right-hand side</li>
<li>Under App Settings scroll down to Output the header only and select True from the dropdown menu</li>
<li>Click on the toggle to turn Batching On</li>
<li>Select File from the Batch by dropdown menu </li>
<li>Click on Select file(s) dropdown menu associated with the Input BAM/SAM/CRAM file</li>
<li>Select all bam files by clicking on the respective checkboxes</li>
<li>Click on the Save selection button on the top right-hand side</li>
<li>Click on the Run button on the top right-hand side</li>
</ul>
<h2 id="step-5-load-the-appropriate-reference-annotation-file">Step 5: Load the appropriate reference annotation file</h2>
<p>In order to count gene features across exons HtSeq-count tool will need a Gene Transfer Format (GTF) file that contains information about each gene from a particular reference genome. In this case, the original BAM files were mapped to the NCBI reference genome canFam3. 
* Download the GTF file located here https://hgdownload.soe.ucsc.edu/goldenPath/canFam3/bigZips/genes/
* Click on canFam3.ncbiRefSeq.gtf.gz
* Locate this file on your local computer and open to unzip
* Navigate back to the CGC and add this file to the project</p>
<h2 id="step-4-count-sequencing-reads-with-htseq-count">Step 4: Count sequencing reads with htseq-count</h2>
<ul>
<li>Click on Apps from the secondary navigation menu </li>
<li>Click HTSeq-count</li>
<li>Click on the Run button from the top right-hand side</li>
<li>Under App Settings select name from the Order dropdown menu</li>
<li>Under App Settings select ignore from the secondary alignments dropdown menu</li>
<li>Under App Settings select ignore from the supplementary alignments dropdown menu</li>
<li>Click on the toggle to turn Batching On</li>
<li>Select File from the Batch by dropdown menu </li>
<li>Click on the Select file(s) button associated with Aligned reads</li>
<li>Select all name sorted bam files from the name_sorted folder</li>
<li>Click on the Select file(s) button associated with Reference annotation file</li>
<li>Select canFam3.ncbiRefSeq.gtf.gz
<img alt="HTSeq-count" src="../rna-seq-images/cgc-htseq-results.png" title="HTSeq-count" /></li>
</ul>
<h2 id="step-5-create-a-csv-file-with-phenotype-data-for-all-samples-for-deseq2">Step 5: Create a csv file with phenotype data for all samples for DESeq2</h2>
<ul>
<li>Click on Files from the menu bar</li>
<li>In the search box type .csv and hit enter</li>
<li>Click on the file manifest which will be named with a series of letters and numbers with a .csv file extension</li>
<li>Click on the Download button to initiate a download to your local machine
<img alt="File Manifest" src="../rna-seq-images/cgc-download-manifest.png" title="File Manifest" /></li>
<li>Open this file in Excel or any similar application</li>
<li>Move the sample_id column so that it is the first column in the file</li>
<li>Delete all rows pertaining to the Index Files with a file_format of bai</li>
<li>Save the file as phenotype_filtered.csv</li>
<li>The file created should be formatted similar to the file shown below
<img alt="phenotype_filtered.csv" src="../rna-seq-images/cgc-phenotype_filtered.png" title="phenotype_filtered.csv" /></li>
<li>Click on the Add files button to expand the dropdown menu and select Your Computer</li>
<li>Click on the Browse files button to add the file created named phenotype_filtered.csv</li>
<li>Click on the green Start upload button</li>
</ul>
<h2 id="step-6-conduct-differential-expression-with-deseq2">Step 6: Conduct differential expression with DESeq2</h2>
<ul>
<li>Click on Apps from the menu bar</li>
<li>Click DESeq2 </li>
<li>Click on the Run button from the top right-hand side</li>
<li>Under App Settings enter Urinary_Bladder_Cancer_DGE as the Analysis title</li>
<li>Under App Settings enter diagnosis as the Covariate of interest</li>
<li>Under App Settings enter 0.01 as the FDR cutoff</li>
<li>Under App Settings enter Healthy Control as the Factor level - reference</li>
<li>Under App Settings enter Bladder Cancer as the Factor level - test</li>
<li>Under App Settings select htseq from the dropdown menu of Quantification tool</li>
<li>Under App Settings select True from the dropdown menu of log2 fold change shrinkage</li>
<li>Under Inputs click on the Select file(s) button associated with Expression data<ul>
<li>Select the counts.tsv files from the Counts folder</li>
</ul>
</li>
<li>Under Inputs click on the Select file(s) button associated with Gene annotation<ul>
<li>Select canFam3.ncbiRefSeq.gtf</li>
</ul>
</li>
<li>Under Inputs click on the Select file(s) button associated with Phenotype data<ul>
<li>Select phenotype_filtered.csv</li>
</ul>
</li>
</ul>
<p><img alt="DESeq2 Parameters" src="../rna-seq-images/cgc-deseq2-parameters.png" title="DESeq2 Parameters" /></p>
<ul>
<li>Click on the green Run button in the upper right-hand corner</li>
<li>When the task has completed successfully inspect the results that should be the same as shown below
<img alt="PCA Plot" src="../rna-seq-images/cgc-pca-plot.png" title="PCA Plot" />
<img alt="Cluster Dendrogram" src="../rna-seq-images/cgc-cluster-dendrogram.png" title="Cluster Dendrogram" />
<img alt="Boxplot" src="../rna-seq-images/cgc-boxplot.png" title="Boxplot" />
<img alt="Summary" src="../rna-seq-images/cgc-analysis-summary.png" title="Summary" /></li>
</ul>

              
            </article>
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" data-md-state="hidden">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../rna_seq_03/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Building a Cohort within the ICDC" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Building a Cohort within the ICDC
            </div>
          </div>
        </a>
      
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.path", "navigation.sections", "navigation.tabs", "navigation.tracking", "navigation.path", "navigation.top", "navigation.expand"], "search": "../../../assets/javascripts/workers/search.2a1c317c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.748e2769.min.js"></script>
      
    
  </body>
</html>